pico/configdriven/runtime/src/main/java/io/helidon/pico/configdriven/runtime/DefaultPicoConfigBeanRegistry.java

```diff
@@ -332,66 +329,36 @@ <T, GCB extends GeneratedConfigBean> void loadConfigBeans(io.helidon.config.Conf
+                                                  + config.key().toString());
+            }
+            GCB baseConfigBean = toConfigBean(config, configuredServiceProvider);
             registerConfigBean(baseConfigBean, null, config, configuredServiceProvider, metaAttributes);
         }
-        mapOfInstanceBasedConfig
-                .forEach((instanceId, configBean) ->
-                                 registerConfigBean(configBean,
-                                                    config.key().toString() + "." + instanceId,
-                                                    config.get(instanceId),
-                                                    configuredServiceProvider,
-                                                    metaAttributes));
     }
 
-    /**
-     * The base config bean must be a root config, and is only available if there is a non-numeric
-     * key in our node list (e.g., "x.y" not "x.1.y").
-     */
-    <T, GCB extends GeneratedConfigBean> GCB maybeLoadBaseConfigBean(io.helidon.config.Config config,
-                                                                     ConfigValue<List<io.helidon.config.Config>> nodeList,
-                                                                     ConfiguredServiceProvider<T, ?> configuredServiceProvider) {
-        boolean hasAnyNonNumericNodes = nodeList.get().stream()
-                .anyMatch(cfg -> toNumeric(cfg.name()).isEmpty());
-        if (!hasAnyNonNumericNodes) {
-            return null;
-        }
-
-        return toConfigBean(config, configuredServiceProvider);
-    }
-
-    /**
-     * These are any {config}.N instances, not the base w/o the N.
-     */
-    <T, GCB extends GeneratedConfigBean> Map<String, GCB> maybeLoadConfigBeans(
-            ConfigValue<List<io.helidon.config.Config>> nodeList,
-            ConfiguredServiceProvider<T, GCB> configuredServiceProvider) {
-        Map<String, GCB> result = new LinkedHashMap<>();
-
-        nodeList.get().stream()
-                .filter(cfg -> toNumeric(cfg.name()).isPresent())
-                .map(ConfigDrivenUtils::safeDowncastOf)
-                .forEach(cfg -> {
-                    String key = cfg.name();
-                    GCB configBean = toConfigBean(cfg, configuredServiceProvider);
-                    Object prev = result.put(key, configBean);
-                    assert (prev == null) : prev + " and " + configBean;
+    <T, CB extends GeneratedConfigBean> void registerChildConfigBeans(io.helidon.config.Config config,
+                                          ConfiguredServiceProvider<T, CB> configuredServiceProvider,
+                                          Map<String, Map<String, Object>> metaAttributes) {
+        config.asNodeList()
+                .ifPresent(list -> {
+                    list.forEach(beanCfg -> {
+                        // use explicit name, otherwise index or name of the config node
```

Comment:
where is the "otherwise index" represented here in the code?

Reply:
Updated the comment